lib/ts/utils.ts

```diff
@@ -207,28 +208,62 @@ export function validateTheStructureOfUserInput(
     }
 }
 
-export async function getAPIVersionFromFileIfExists(): Promise<string | null> {
+export async function getDataFromFileIfExists<T>(filePath: string): Promise<T | undefined> {
     try {
-        let apiVersionFromFile = await new Promise<Buffer>((resolve, reject) => {
-            readFile(API_VERSION_FILE_PATH, (err, data) => {
+        let dataFromFile = await new Promise<Buffer>((resolve, reject) => {
+            readFile(filePath, (err, data) => {
                 if (err !== undefined && err !== null) {
                     reject(err);
                 }
                 resolve(data);
             });
         });
-        return apiVersionFromFile.toString();
+        return JSON.parse(dataFromFile.toString());
     } catch (err) {
-        return null;
+        return undefined;
     }
 }
 
-export async function storeAPIVersionInFile(apiVersion: string) {
+export async function storeIntoTempFile(filePath: string, data: any) {
+    try {
+        await createTempDirIfNotAlreadyExists();
+        await new Promise(async (resolve, reject) => {
+            writeFile(filePath, JSON.stringify(data), (err) => {
+                resolve(undefined);
+            });
+        });
+    } catch (err) {}
+}
+
+export async function createTempDirIfNotAlreadyExists() {
+    // we allow this to throw error as it will be called from the try block of storeIntoTempFile function
+    await new Promise((resolve, reject) => {
+        // will have no effect if the directory already exists
+        mkdir("/temp", (err) => {
+            if (err !== undefined && err !== null) {
+                reject(err);
+            }
+            resolve(undefined);
+        });
+    });
+}
+
+async function removeFile(filePath: string) {
     try {
         await new Promise((resolve, reject) => {
-            writeFile(API_VERSION_FILE_PATH, apiVersion, {}, () => {
+            unlink(filePath, (err) => {
+                if (err !== undefined && err !== null) {
+                    reject(err);
+                }
                 resolve(undefined);
             });
         });
     } catch (err) {}
 }
+
+export async function removeTempFiles() {
```

Comment:
Change the name of this function to be `removeServerlessCache`

Reply:
done

Reply:
Not done.. code is still `removeTempFiles`.